e2e/LongHaul/service/IotHub.cs

```diff
@@ -111,6 +111,10 @@ await Task
                             {
                                 _logger.Trace($"Operations on [{deviceId}] have been canceled as the device goes offline.", TraceSeverity.Information);
                             }
+                            catch (Exception ex)
+                            {
+                                _logger.Trace($"Service app failed with exception {ex}", TraceSeverity.Error);
```

Comment:
Please add a `ToString()` override to our exception classes in iot hub (consider dps) to include ErrorCode, etc.

like:
Message
ErrorCode={}, StatusCode={}, IsTransient={}, TrackingId={}
CallStack